File: support_tools/swigwin-3.0.12/Examples/test-suite/python/using_extend_runme.py
from using_extend import *

f = FooBar()
if f.blah(3) != 3:
    raise RuntimeError, "blah(int)"

if f.blah(3.5) != 3.5:
    raise RuntimeError, "blah(double)"

if f.blah("hello") != "hello":
    raise RuntimeError, "blah(char *)"

if f.blah(3, 4) != 7:
    raise RuntimeError, "blah(int,int)"

if f.blah(3.5, 7.5) != (3.5 + 7.5):
    raise RuntimeError, "blah(double,double)"


if f.duh(3) != 3:
    raise RuntimeError, "duh(int)"

File: support_tools/swigwin-3.0.12/Examples/test-suite/ruby/using_extend_runme.rb
#!/usr/bin/env ruby
#
# Put description here
#
# 
# 
# 
#

require 'swig_assert'

require 'using_extend'

include Using_extend

f = FooBar.new
if f.blah(3) != 3
  raise RuntimeError,"blah(int)"
end

if f.blah(3.5) != 3.5
  raise RuntimeError,"blah(double)"
end

if f.blah("hello") != "hello"
  raise RuntimeError,"blah(char *)"
end

if f.blah(3,4) != 7
  raise RuntimeError,"blah(int,int)"
end

if f.blah(3.5,7.5) != (3.5+7.5)
  raise RuntimeError,"blah(double,double)"
end

